### PR TITLE
WIP: Inventory CRUD API

### PR DESCRIPTION
**Pull request by lemilonkh**

### CODE REVIEW

File: app/models/index.js [hunk 1]
```diff
@@ -0,0 +1,43 @@
...
+
+let sequelize;
+if (config.use_env_variable) {
+  sequelize = new Sequelize(process.env[config.use_env_variable], config);
+} else {
+  sequelize = new Sequelize(config.database, config.username, config.password, config);
+}
+
+fs
+  .readdirSync(__dirname)
```
Comment (evanp) [inline app/models/index.js:20]: So, you're reading all the models in the directory?

How many models are we expecting, such that this is easier than just listing the models directly?

File: app/next.config.js [hunk 2]
```diff
@@ -1,4 +1,14 @@
 /** @type {import('next').NextConfig} */
-const nextConfig = {}
+const nextConfig = {
+  serverRuntimeConfig: {
```
Comment (evanp) [inline app/next.config.js:3]: Great. We want our config to come from the environment vars

File: app/src/util/api.ts [hunk 3]
```diff
@@ -2,10 +2,18 @@ import createHttpError from "http-errors";
...
+
 export function apiHandler(handler: (req: Request) => Promise<NextResponse>) {
   return async (req: Request) => {
     try {
+      if (!db.initialized) {
+        await db.initialize();
+      }
+
       // TODO JWT authentication logic here
+      // await jwtMiddleware(req);
```
Comment (evanp) [inline app/src/util/api.ts:15]: So, we'll need both session and JWT auth.

**Review by evanp — COMMENTED** [review]
Overall pretty good. I'm concerned about the complicated file loading; it feels too clever. Is it Sequelize boilerplate?

File: app/models/index.js (hunk 1 above)
Comment (lemilonkh) [reply app/models/index.js:20]: This is what sequelize generates when being setup, we can probably get rid of this and use our own initialization file (src/models/index.ts). I'm happy to add them manually in src/models/index.ts instead, where we would have the option to reference them and create the relations etc.

We'll just have to change the options for sequelize so it reads the models from there.

File: app/src/util/api.ts (hunk 3 above)
Comment (lemilonkh) [reply app/src/util/api.ts:15]: Yes, to support a proper (immediate) logout I would suggest using redis for storing the sessions.

**Comment by lemilonkh** [comment]
@evanp yes, this is what `npx sequelize-cli init` generates... I'll get rid of it and load the models manually.

File: app/models/index.js (hunk 1 above)
Comment (evanp) [reply app/models/index.js:20]: If it's the generated version, let's just use it!

**Pull request merged**